### PR TITLE
Add new RRSwISC6to18E3r3 ocean and sea-ice mesh

### PR DESCRIPTION
**Pull request by xylar**
Long name: RRSwISC6to18L80E3SMv3r3

This RRS (Rossby-radius scaled) mesh has:
* 6 km resolution near the poles
* 18 km resolution at the equator

This is a proposed E3SM v3 (E3) high resolution (near-eddy-resolving) mesh.  This is revision 3 (r3) of the mesh, since we caught and fixed the issues with Greenland topography (seen in earlier revisions of EC and Icos meshes) before any substantial testing was performed.

It differs from https://github.com/E3SM-Project/E3SM/pull/6060 (RRSwISC6to18E3r1) mostly in having a the `landIceMask` bug fix from https://github.com/MPAS-Dev/compass/pull/73.

The mesh was created using [compass](https://github.com/MPAS-Dev/compass), specifically this PR: 
https://github.com/MPAS-Dev/compass/pull/747
A compass tag will be created for the mesh as soon as the PR is merged.

The mesh and forthcoming simulation results will be reviewed here:
https://acme-climate.atlassian.net/wiki/spaces/OO/pages/4034658887/Review+RRSwISC6to18E3r3

G- and B-case simulations will begin shortly and analysis will be posted here and on the review page as soon as it is available.

### CODE REVIEW

**Comment by xylar** [comment]
@jonbob, I'm nearly done generating the support files for this mesh in Compass (the sea-ice graph files have taken more then 24 hours). I'll sync them over to `/lcrc/group/e3sm/data/inputdata` tomorrow morning, at which point you can make the appropriate mapping and domain files whenever you have time.  I'll keep you posted.

**Comment by xylar** [comment]
@jonbob, the Compass `ocn` and `ice` files for this mesh have been synced over to `/lcrc/group/e3sm/data/inputdata` as promised.

**Comment by xylar** [comment]
@proteanplanet, I'm pretty sure you won't have time to review this mesh like you did #6060.  I think that's fine because they're very nearly the same except for the bug fix for `landIceMask`.

**Comment by xylar** [comment]
@vanroekel, please have a look when you can.

@eclare108213 and @erinethomas, it would be great to have sea-ice folks give this a look.  Since @proteanplanet has been doing it in the past but he won't be around, maybe one or both of you can take a look.  Presumably, you can wait until there's a G- or B-case simulation to look at and then look for any trouble spots (e.g. cells where ice accumulates more than expected or isolated cells on the B-grid).

**Review by proteanplanet — APPROVED** [review]
I have thoroughly checked this mesh as [RRSwISC6to18E3r1](https://github.com/E3SM-Project/E3SM/pull/6060) and approved it based on acuity of coastal bathymetry, integrity of Antarctic ice shelf cavities, and narrow passages globally, including in the Arctic.  The fix being applied as RRSwISC6to18E3r3 does not change the algorithm used to generate the features I reviewed in that mesh.  Therefore I also approve this mesh based on the previously-reviewed bathymetric integrity of the mesh.  I will generate mesh figures appropriate for publications and usable by the E3SM team in 2024.

**Comment by jonbob** [comment]
Thanks @xylar -- I'll start making the other files to support this now

**Comment by jonbob** [comment]
Added mapping and domain files to support the following resolutions:
* T62_RRSwISC6to18E3r3
* TL319_RRSwISC6to18E3r3
* ne30pg2_RRSwISC6to18E3r3
* ne30pg2_r05_RRSwISC6to18E3r3
* ne120pg2_r05_RRSwISC6to18E3r3
* ne120pg2_r0125_RRSwISC6to18E3r3

Complete support will require more testing

**Comment by jonbob** [comment]
new files are staged in /lcrc/group/e3sm/data/inputdata/ for testing and not yet part of the inputdata repo

**Review by vanroekel — APPROVED** [review]
Approving based on visual inspection and testing from @jonbob and @fspereira1

**Comment by erinethomas** [comment]
waiting for new simulation results before submitting my approval.

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml
Comment (xylar) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:52]: This will likely need to be reduced:
```suggestion
<config_dt ocn_grid="RRSwISC6to18E3r3">'00:05:00'</config_dt>
```

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml
Comment (xylar) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:488]: This will likely need to be reduced:
```suggestion
<config_btr_dt ocn_grid="RRSwISC6to18E3r3">'0000_00:00:10'</config_btr_dt>
```

**Comment by xylar** [comment]
Closing in favor of https://github.com/E3SM-Project/E3SM/pull/6143

**Pull request closed without merging**
